src/globus_cli/exception_handling/hooks.py

```diff
@@ -25,17 +25,31 @@ def _pretty_json(data: dict, compact=False) -> str:
     exit_status=4,
 )
 def handle_internal_auth_requirements(exception: CLIAuthRequirementsError) -> None:
-    if not (exception.gare and exception.gare.authorization_parameters.required_scopes):
+    gare = exception.gare
+    if not gare:
         click.secho(
             "Fatal Error: Unsupported internal auth requirements error!",
             bold=True,
             fg="red",
         )
         click.get_current_context().exit(255)
 
-    required_scopes = exception.gare.authorization_parameters.required_scopes
-
-    _concrete_consent_required_hook(exception.message, required_scopes)
+    required_scopes = gare.authorization_parameters.required_scopes
+    if required_scopes:
+        ret = _concrete_consent_required_hook(exception.message, required_scopes)
+        if ret is not None:
+            click.get_current_context().exit(ret)
```

Comment:
This is the reverse of the

```python
if ret is None:
    return None
```

pattern that we've discussed before. 😆 This code would be clearer if it read as:

```suggestion
        if ret is not None:
            click.get_current_context().exit(255)
```

Don't do that! But I've commented below that the return value of `_concrete_consent_required_hook` is not an arbitrary int -- it's 255, always.

Reply:
It _happens_ to be `Literal[255] | None` _today_. But the defined interface is not that.